### PR TITLE
SAK-48845 Portal improve iframe dark mode compatibility

### PR DESCRIPTION
**Pull request by kunaljaykam**
https://sakaiproject.atlassian.net/browse/SAK-48845

### CODE REVIEW

File: library/src/webapp/js/portal/portal.theme.switcher.js [hunk 1]
```diff
@@ -34,13 +34,42 @@ portal.setPortalThemeUserPref = theme => {
...
+  } else if (document.documentElement.classList.contains(portal.lightThemeClass)) {
+    return portal.lightThemeClass;
+  }
+  return portal.defaultThemeClass;
+};
+
+portal.updateIframeTheme = (themeClass) => {
+
+  const iframes = document.querySelectorAll('iframe');
+  iframes.forEach(iframe => {
```
Comment (adrianfish) [inline library/src/webapp/js/portal/portal.theme.switcher.js:50]:
```suggestion
  document.querySelectorAll('iframe').forEach(iframe => {
```

File: library/src/webapp/js/portal/portal.theme.switcher.js [hunk 2]
```diff
@@ -51,9 +80,21 @@ portal.enableLightTheme = () => {
...
   portal.addCssClassToMarkup(portal.lightThemeClass);
   portal.setPortalThemeUserPref(portal.lightThemeClass);
+  portal.updateIframeTheme(portal.lightThemeClass);
   portal.darkThemeSwitcher.title = portal.i18n.theme_switch_to_dark;
 };
 
+document.addEventListener('DOMContentLoaded', function() {
+  
+  if (portal.userThemeAutoDetectDark && portal.user.id && portal.userTheme === portal.defaultThemeClass) {
+      const themeToApply = portal.isOsDarkThemeSet() ? portal.darkThemeClass : portal.lightThemeClass;
```
Comment (adrianfish) [inline library/src/webapp/js/portal/portal.theme.switcher.js:90]: You're using 4 spaces here, but 2 in other parts of this pr.

**Review by adrianfish — APPROVED** [review]
Couple of tiny things, that's all. Thanks Kunal.

**Comment by kunaljaykam** [comment]
Thanks!

**Pull request merged**